### PR TITLE
[1314] Add funding field to courses

### PR DESCRIPTION
**Pull request by tvararu**
### Context

We need to expose additional fields for use in the frontend.

### Changes proposed in this pull request

- Add funding field

### Guidance to review

🚢

### CODE REVIEW

**Comment by tvararu** [comment]
[Course basic details: apprenticeship + fee or salary](https://trello.com/c/iyS7eOww/1314-course-basic-details-apprenticeship-fee-or-salary)

**Comment by dankmitchell** [comment]
Alternative approach could be a `fee` AND `salary` attribute that return a boolean

**Comment by dankmitchell** [comment]
And perhaps the `program_type` could be an enum to display more helpful data like we done in a few other places

**Comment by dankmitchell** [comment]
Feel free to ignore the above comments. You’ve got my approval 💪

File: spec/requests/api/v2/providers/courses_spec.rb [hunk 1]
```diff
@@ -17,6 +17,7 @@
            study_mode: :full_time,
            subject_count: 0,
            subjects: [course_subject],
+           program_type: 'TA',
```
Comment (defong) [inline spec/requests/api/v2/providers/courses_spec.rb]: Can we add the known fields as part of the course factory as a trait, along with a default one?
It be nice capture `random chars` to actual meanings

Comment (tvararu) [reply spec/requests/api/v2/providers/courses_spec.rb]: Done, good suggestion

File: spec/factories/courses.rb [hunk 2]
```diff
@@ -27,6 +27,7 @@
     sequence(:course_code) { |n| "C#{n}D3" }
     name { Faker::ProgrammingLanguage.name }
     qualification { :pgce_with_qts }
+    program_type { 'HE' }
```
Comment (defong) [inline spec/factories/courses.rb]: maybe 
```
...
qualification { :pgce_with_qts }
with_he
association(:provider)
...
```

    trait :with_he do
      program_type { 'HE' }
    end

**Pull request merged**